src/hal/drivers/mesa-hostmot2/sserial.c

```diff
@@ -2016,7 +2016,9 @@ int hm2_sserial_read_pins(hm2_sserial_remote_t *chan){
                 break;
             case LBP_BOOLEAN:
                 *pin->boolean = (buff != 0);
-                *pin->boolean2 = (buff == 0);
+                if(conf->DataDir == LBP_IN){
+                    *pin->boolean2 = (buff == 0);
+                }
```

Comment:
Do you want to specify a default value here for when DataDir is not LBP_IN, like you did in the hunk above?

Reply:
that is not required, as boolean2 is not created in the other case: https://github.com/LinuxCNC/linuxcnc/blob/master/src/hal/drivers/mesa-hostmot2/sserial.c#L1310-L1316
for the other direction it is required, because in case of outputs its the invert pin.
the bug was introduces here: https://github.com/LinuxCNC/linuxcnc/commit/31029ef2166d1f52745baa5b458358dd72fe91d4

Reply:
Cool.